sdk/core/Azure.Core/perf/SerializationBenchmark/BimodalRepro.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Reflection;
+using Azure.Core.Serialization;
+using BenchmarkDotNet.Attributes;
+
+namespace Azure.Core.Perf
+{
+    [MemoryDiagnoser]
+    [InProcess]
+    public class BimodalRepro
+    {
+        private ModelSerializerOptions _options;
+        private BinaryData _data;
+
+        [GlobalSetup]
+        public void SetUp()
+        {
+            string json = File.ReadAllText(Path.Combine(Directory.GetParent(Assembly.GetExecutingAssembly().Location).FullName, "SerializationBenchmark", "TestData", "AvailabilitySetData.json"));
+            _data = BinaryData.FromString(json);
+            _options = ModelSerializerOptions.AzureServiceDefault;
+        }
+
+        [Benchmark]
+        public void Bimodal()
+        {
+            using var content = new MultiBufferRequestContent();
```

Comment:
I've checked `MultiBufferRequestContent` ctor and it's very simple (no allocations). We would need to use a profiler to find out why it's bimodal. Sample perf investigation: https://adamsitnik.com/Sample-Perf-Investigation/

Is it also bimodal when run without `InProcess` toolchain?

Reply:
yes its still bimodal after applying all of the changes you suggested.

```
-------------------- Histogram --------------------
[ 4.815 ns ;  9.355 ns) | @@@@@@@@@@@@@
[ 9.355 ns ; 14.099 ns) |
[14.099 ns ; 18.958 ns) | @@@@@@@
---------------------------------------------------
```